staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -431,10 +511,27 @@ func NewDiffer(from, to string) (*Differ, error) {
 
 // Diff diffs to versions of a specific object, and print both versions to directories.
 func (d *Differ) Diff(obj Object, printer Printer) error {
-	if err := d.From.Print(obj, printer); err != nil {
+	from, err := d.From.getObject(obj)
+	if err != nil {
+		return err
+	}
+	to, err := d.To.getObject(obj)
+	if err != nil {
+		return err
+	}
+
+	// Mask secret values if object is V1Secret
+	if gvk := obj.GroupVersionKind(); gvk.Version == "v1" && gvk.Kind == "Secret" {
```

Comment:
To a question here and pardon if this is too much basic :)

the obj.Live() already have a GroupVersionKind method, like:

obj.Live().GetObjectKind().GroupVersionKind()

So instead of adding this into the interface, wouldn't be better to only call this from the already Live() method?

Thanks!

Reply:
Oh yes, this is something I considered too but I'm not sure if it was ok to get it from either the `Live`/`Merged` object, as opposed to getting from the original object. If this is ok, I can remove the GVK-related additions.

Reply:
@apelisse WDYT?